lib/publiccloud/ec2.pm

```diff
@@ -17,13 +17,58 @@ use testapi;
 
 has ssh_key      => undef;
 has ssh_key_file => undef;
+has credentials  => undef;
+
+sub create_credentials {
+    my ($self) = @_;
+
+    if (!defined($self->credentials)) {
+        my $credentials = $self->do_rest_api('/ec2/users', method => 'post');
+        die("Missing key") if (!defined($credentials->{keys}));
```

Comment:
good enough for POC but later we need to think about getting proper error here : 
"Service down "
"Internal Server Error" 
404  or something like this . 
"Missing key" does not help much to person who will review failure and try to figure out how to fix it

Reply:
Agree, the 404 or internal error should appear if the http request fail. This could be something like, `unknown credential format retrieved`.